### PR TITLE
[flang][OpenMP] Handle `omp.private` in `FirOpBuilder::getAllocaBlock()`

### PR DESCRIPTION
**Pull request by ergawy**
Fixes a crash uncovered by [pr89651](https://github.com/llvm/llvm-test-suite/blob/main/Fortran/gfortran/regression/gomp/pr89651.f90) in the test suite.

Fixes a crash caused by missing handling of `omp.private` ops in `FirOpBuilder::getAllocaBlock()`.

### CODE REVIEW

File: flang/test/Lower/OpenMP/delayed-privatization-allocatable-firstprivate.f90 [hunk 1]
```diff
@@ -34,3 +37,17 @@ subroutine delayed_privatization_allocatable
 ! CHECK-NEXT:    %[[ORIG_BASE_LD:.*]] = fir.load %[[ORIG_BASE_ADDR]]
 ! CHECK-NEXT:    hlfir.assign %[[ORIG_BASE_LD]] to %[[PRIV_BASE_BOX]] temporary_lhs
 ! CHECK-NEXT:  }
+
+! RUN: %flang -c -fopenmp -mmlir --openmp-enable-delayed-privatization \
+! RUN:   -o - %t/test_compilation_to_obj.f90
+
+!--- test_compilation_to_obj.f90
```
Comment (tblah) [inline flang/test/Lower/OpenMP/delayed-privatization-allocatable-firstprivate.f90:44]: nit: Please could you add some minimal CHECK lines to see that we did actually build this file and successfully produce output

**Review by tblah — APPROVED** [review]
Code change LGTM. Thanks for the bug fix

File: flang/test/Lower/OpenMP/delayed-privatization-allocatable-firstprivate.f90 (hunk 1 above)
Comment (ergawy) [reply flang/test/Lower/OpenMP/delayed-privatization-allocatable-firstprivate.f90:46]: Done.

File: flang/lib/Optimizer/Builder/FIRBuilder.cpp [hunk 2]
```diff
@@ -252,7 +252,8 @@ mlir::Block *fir::FirOpBuilder::getAllocaBlock() {
               .getParentOfType<mlir::omp::OutlineableOpenMPOpInterface>()) {
     return ompOutlineableIface.getAllocaBlock();
   }
-  if (getRegion().getParentOfType<mlir::omp::DeclareReductionOp>())
+  if (getRegion().getParentOfType<mlir::omp::DeclareReductionOp>() ||
+      getRegion().getParentOfType<mlir::omp::PrivateClauseOp>())
```
[the remark this answers is not in the document]
Comment (ergawy) [reply flang/lib/Optimizer/Builder/FIRBuilder.cpp]: Makes sense. I updated the PR to reuse the `RecipeInterface` for both OpenMP and OpenACC.

File: mlir/include/mlir/Dialect/OpenACCMPCommon/Interfaces/OpenACCMPOpsInterfaces.td [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
-#ifndef OPENACC_OPS_INTERFACES
-#define OPENACC_OPS_INTERFACES
+#ifndef OPENACC_MP_COMMON_OPS_INTERFACES
+#define OPENACC_MP_COMMON_OPS_INTERFACES
 
 include "mlir/IR/OpBase.td"
 
 def RecipeInterface : OpInterface<"RecipeInterface"> {
     let description = [{
       OpenACC operations with one or more regions holding executable code.
```
Comment (tblah) [inline mlir/include/mlir/Dialect/OpenACCMPCommon/Interfaces/OpenACCMPOpsInterfaces.td:20]: nit
```suggestion
      OpenACC/OpenMP operations with one or more regions holding executable code.
```

**Pull request merged**